wallet-gateway/remote/src/cache/network-cache.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { LRUCache } from 'typescript-lru-cache'
+import {
+    Store,
+    Network,
+    VerifiedNetwork,
+} from '@canton-network/core-wallet-store'
+import { Logger } from 'pino'
+
+export const DEFAULT_MAX_CACHE_SIZE = 100
+export const DEFAULT_ENTRY_EXPIRATION_TIME = 10 * 60 * 1000
+
+const NetworkCache = new LRUCache<string, VerifiedNetwork>({
+    maxSize: DEFAULT_MAX_CACHE_SIZE,
+    entryExpirationTimeInMS: DEFAULT_ENTRY_EXPIRATION_TIME,
+})
+
+export class NetworkCacheStore {
+    constructor(
+        private readonly store: Store,
+        private readonly logger: Logger
+    ) {}
```

Comment:
Do you need all this? I'm wondering why not decouple the cache from the underlying store?

The network cache can just store `{ networkId: string, verified: boolean }`, and only needs a getter and setter.

In the controller you retrieve the latest status from the store directly, and merge based on ID to include the verified status